py/desispec/pipeline/tasks/qadata.py

```diff
@@ -0,0 +1,134 @@
+    """Class containing the properties of a sky fit task.
+    """
+    def __init__(self):
+        super(TaskQAData, self).__init__()
+        # then put int the specifics of this class
+        # _cols must have a state
+        self._type = "qadata"
+        self._cols = [
+            "night",
+            "band",
+            "spec",
+            "expid",
+            "state"
+        ]
+        self._coltypes = [
+            "integer",
+            "text",
+            "integer",
+            "integer",
+            "integer"
+        ]
+        # _name_fields must also be in _cols
+        self._name_fields  = ["night","band","spec","expid"]
+        self._name_formats = ["08d","s","d","08d"]
+
+    def _paths(self, name):
+        """See BaseTask.paths.
+        """
+        props = self.name_split(name)
+        camera = "{}{}".format(props["band"], props["spec"])
+        return [ findfile("qa_data", night=props["night"], expid=props["expid"],
+            camera=camera) ]  # Add qaprod_dir here to have QA land somewhere else?
+
+    def _deps(self, name, db, inputs):
+        """See BaseTask.deps.
+        """
+        from .base import task_classes
+        props = self.name_split(name)
+        deptasks = {
+            "cframe" : task_classes["cframe"].name_join(props),
+        }
+        return deptasks
+
+    def _run_max_procs(self, procs_per_node):
+        """See BaseTask.run_max_procs.
+        """
+        return 1
+
+
+    def _run_time(self, name, procs_per_node, db=None):
+        """See BaseTask.run_time.
+        """
+        return 2 # less than a minute (for the simple sky fit)
+
+
+    def _run_defaults(self):
+        """See BaseTask.run_defaults.
+        """
+        opts = {}
+        return opts
+
+
+    def _option_list(self, name, opts):
+        """Build the full list of options.
+
+        This includes appending the filenames and incorporating runtime
+        options.
+        """
+        from .base import task_classes, task_type
+
+        props = self.name_split(name)
+        options = {}
+        options["frame_file"] = task_classes["extract"].paths(
+            task_classes["extract"].name_join(props))
+
+        options.update(opts)
+        return option_list(options)
+
+    def _run_cli(self, name, opts, procs, db):
+        """See BaseTask.run_cli.
+        """
+        entry = "desi_qa_frame"
+        optlist = self._option_list(name, opts)
+        com = "{} {}".format(entry, " ".join(optlist))
+        return com
+
+    def _run(self, name, opts, comm, db):
+        """See BaseTask.run.
+        """
+        from ...scripts import qa_frame
+        optlist = self._option_list(name, opts)
+        args = qa_frame.parse(optlist)
+        qa_frame.main(args)
+        return
+
+    def postprocessing(self, db, name, cur):
+        """For successful runs, postprocessing on DB"""
+        # run getready on all fierflatnight with same night,band,spec
+        pass
+        '''
```

Comment:
You can remove this commented block- qadata tasks will not have further tasks that depend on them...

Reply:
This comment is highlighting the wrong code- I meant to remove the commented block in postprocessing()...